[14.0][MIG] account_tax_unece

alexis-via opened the pull request.

alexis-via committed:
Add modules base_unece, account_tax_unece, account_payment_unece and product_uom_unece

alexis-via committed:
Port *unece* modules to odoo 10

astirpe committed:
Add unece type and category to tax template

Commit:
[10.0] Add field 'UNECE Due date' on taxes and corresponding nomenclature (#26)

astirpe committed:
[11.0][MIG] account_tax_unece: porting to V11

lk-eska committed:
[MIG] account_tax_unece: Migration to 12.0

alexis-via committed:
Add unece_due_date_id on account.tax.template

pedrobaeza committed:
[IMP] account_tax_unece: README by fragments

oca-travis committed:
Ocabot updates

[UPD] Update account_tax_unece.pot

[ADD] icon.png

[UPD] README.rst

astirpe committed:
[13.0][MIG] account_tax_unece

oca-travis committed:
[UPD] Update account_tax_unece.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
pre-commit update

alexis-via committed:
[MIG] account_tax_unece from v13 to v14

Remove field unece_due_date_id because we'll use tax_exigibility
Remove test for invoice creation (will be moved to module account_einvoice_generate from OCA/edi)

alexis-via commented:
Requires #98

pedrobaeza commented on account_tax_unece/__manifest__.py:
```diff
@@ -1,22 +1,23 @@
+    "version": "14.0.1.0.0",
     "category": "Accounting & Finance",
     "license": "AGPL-3",
+    "development_status": "Production/Stable",
```
I would say this one can be already `Mature`

pedrobaeza commented on account_tax_unece/data/unece_tax_categ.xml:
```diff
@@ -20,57 +20,62 @@ Source : https://www.unece.org/fileadmin/DAM/trade/untdid/d16b/tred/tred5305.htm
-            name="description"
-        >Duty associated with shipment of goods is paid by the supplier; customer receives goods with duty paid.</field>
+        <field name="description">
+            Duty associated with shipment of goods is paid by the supplier; customer
```
Are all of these changes done by pre-commit? If so, they must be in the pre-commit commit, not here.

alexis-via replied:
What is a pre-commit commit ? Does it mean that you would have prefered to see in a separate commit ?

pedrobaeza replied:
Please check https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch where migration steps including pre-commit are included

astirpe replied:
It is possible to skip the automatic reformatting of a particular file by adding its path in the `.prettierignore`, like we did here: https://github.com/OCA/l10n-netherlands/blob/13.0/.prettierignore

pedrobaeza replied:
The reformatting is good and should be done here, but on its separate commit according the procedure.

astirpe replied:
The reformatting is not good.

Here for example we have:

```xml
        <field name="description">
            Duty associated with shipment of goods is paid by the supplier
```

which however does not behave the same as:


```xml
        <field
           name="description"
          >Duty associated with shipment of goods is paid by the supplier
```

If you want to realize it, open the unece.code.list tree view: you will notice that the text in the description column is misaligned in most of the records.

Since the automatic reformatting is forcing such unwanted changes, just include the path of the files in the .prettierignore so that the reformatting will be not executed.

pedrobaeza replied:
OK, then it's good to know and indeed it affects in this case.

@Yajo is this something related with ignore-whitespace things?

yajo replied:
Fix in https://github.com/OCA/oca-addons-repo-template/pull/35.

pedrobaeza reviewed (changes requested):
It seems there are changes in the migration commit that belongs to pre-commit. Also, the pre-commit commit message is not the one stated in the migration guide.

astirpe reviewed (changes requested):
See my comment https://github.com/OCA/community-data-files/pull/99#discussion_r510832164

rvalyi commented:
@alexis-via so now that the pre-commit hook was fixed with a less aggressive prettier XML plugin, (https://github.com/OCA/oca-addons-repo-template/issues/34) I think you should redo (and amend) your PR and it should fix the wrongly reformatted XML code mentioned above.

As for the pre-commit thing, aside from https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch you can also read an introduction here https://odoo-community.org/blog/the-oca-blog-1/post/black-isort-pre-commit-97
It's easy to set up on your local PC with pre-commit install / pre-commit uninstall and when last glitches will be sorted out at the OCA level it may really improve the average OCA code readability.

alexis-via committed:
account_tax_unece: Restore the inital XML formatting

yajo commented:
I think what you should do is this instead.

```bash
git reset --hard 88af9af61af15817476cd84a67198f9abc2a946b

# Assumming `upstream` is the remote that points to OCA's repo
git pull upstream 14.0 --rebase

pre-commit run -a
git commit -am '[IMP] account_tax_unece: run pre-commit'

git cherry-pick 34c423e2f9fdd2f149dad17805cd8ca36b72b728

# Now fix conflicts, if there are any (there'll be). After that's done:
git cherry-pick --continue
git push --force
```

I'm sorry this problem hit you

alexis-via committed:
account_tax_unece: add speeddict methods

Improve code for unece_due_date_code, to make it easier to inherit if needed for some countries

phuctranfxvn commented:
Hello, I see that you exclude the [test_account_invoice](https://github.com/OCA/community-data-files/blob/13.0/account_tax_unece/tests/test_account_invoice.py) from your test case. Did you do it accidentally or for a specific reason ?
I think you should include the setup/account_tax_unece in your PR also.

StefanRijnhart commented:
@alexis-via will you follow-up? Or if you'd like, I can post a superseding PR.

alexis-via commented:
@StefanRijnhart Don't hesitate to supersede the PR :)

StefanRijnhart commented:
Thanks! Continued here: https://github.com/OCA/community-data-files/pull/103
